src/Simplex/Chat/Library/Commands.hs

```diff
@@ -2025,18 +2027,39 @@ processChatCommand' vr = \case
           updateCIGroupInvitationStatus user g CIGISAccepted `catchChatError` (toView . CRChatError (Just user))
           pure $ CRUserAcceptedGroupSent user g {membership = membership {memberStatus = GSMemAccepted}} Nothing
         Nothing -> throwChatError $ CEContactNotActive ct
-  APIAcceptMember groupId gmId role -> withUser $ \user -> do
+  APIAcceptMember groupId gmId role -> withUser $ \user@User {userId} -> do
     (gInfo, m) <- withFastStore $ \db -> (,) <$> getGroupInfo db vr user groupId <*> getGroupMemberById db vr user gmId
-    assertUserGroupRole gInfo GRAdmin
-    when (memberStatus m /= GSMemPendingApproval) $ throwChatError $ CECommandError "member is not pending approval"
-    case memberConn m of
-      Just mConn -> do
-        let msg = XGrpLinkAcpt role
-        void $ sendDirectMemberMessage mConn msg groupId
-        m' <- withFastStore' $ \db -> updateGroupMemberAccepted db user m role
-        introduceToGroup vr user gInfo m'
-        pure $ CRJoinedGroupMember user gInfo m'
-      _ -> throwChatError CEGroupMemberNotActive
+    assertUserGroupRole gInfo GRModerator
+    case memberStatus m of
+      GSMemPendingApproval | memberCategory m == GCInviteeMember -> do -- only host can approve
```

Comment:
it may be better to have different APIs - approve/accept

Reply:
where approve can trigger admission (maybe admit is a better term than accept).

Reply:
we do send the same protocol message though. but other logic is completely different, yes

Reply:
maybe we should add protocol message too, then, XGrpLinkAdmit, and remove optional memberId from XGrpLinkAcpt (it will return to being only for invitee)